src/libaktualizr-posix/ipuptanesecondary.cc

```diff
@@ -163,37 +220,136 @@ data::ResultCode::Numeric IpUptaneSecondary::install(const std::string& target_n
   return static_cast<data::ResultCode::Numeric>(r->result);
 }
 
-Manifest IpUptaneSecondary::getManifest() const {
-  LOG_DEBUG << "Getting the manifest from secondary with serial " << getSerial();
+data::ResultCode::Numeric IpUptaneSecondary::install_v2(const Uptane::Target& target) {
+  data::ResultCode::Numeric data_delivery_result = data::ResultCode::Numeric::kDownloadFailed;
+
+  LOG_INFO << "Invoking an image delivery of the target  " << target.filename() << " to/on the Secondary "
+           << getSerial();
+
+  if (target.IsOstree()) {
+    data_delivery_result = downloadOstreeRev(target);
+  } else {
+    data_delivery_result = uploadFirmware(target);
+  }
+
+  if (data::ResultCode::Numeric::kOk != data_delivery_result) {
+    return data_delivery_result;
+  }
+
+  LOG_INFO << "Invoking an installation of the target " << target.filename() << " on the Secondary " << getSerial();
```

Comment:
And similarly here "Instructing Secondary <xyz> to install target <abc>".

Reply:
ok